examples/todomvc/src/element.rs

```diff
@@ -0,0 +1,235 @@
+                                            selector
+                                        ).as_str());
+*/
+                                        for i in 0..potential_elements.length() {
+                                            if let Some(el) = potential_elements.get(i) {
+                                                if target_element.is_equal_node(Some(&el)) {
+                                                    has_match = true;
+                                                }
+                                                break;
+                                            }
+                                        }
+
+                                        if has_match {
+                                            handler(event);
+                                        }
+                                    }
+                                }
+                            }
+                        }) as Box<FnMut(_)>);
+
+                        dyn_el.add_event_listener_with_callback_and_bool(
+                            //element_et.add_event_listener_with_callback_and_bool(
+                            event,
+                            cb.as_ref().unchecked_ref(),
+                            use_capture,
+                        );
+                        cb.forget(); // TODO cycle collect
+                    }
+                }
+            }
+            self.el = Some(el);
+        }
+    }
+
+    /// Find child `Element`s from this node
+    pub fn qs_from(&mut self, selector: &str) -> Option<Element> {
+        let mut found_el = None;
+        if let Some(el) = self.el.take() {
+            found_el = Some(Element {
+                el: el.query_selector(selector).ok()?,
+            });
+            self.el = Some(el);
+        }
+        found_el
+    }
+
+    /// Sets the inner HTML of the `self.el` element
+    pub fn set_inner_html(&mut self, value: String) {
+        if let Some(el) = self.el.take() {
+            el.set_inner_html(&value);
+            self.el = Some(el);
+        }
+    }
+
+    /// Sets the text content of the `self.el` element
+    pub fn set_text_content(&mut self, value: String) {
+        if let Some(el) = self.el.take() {
+            if let Some(node) = wasm_bindgen::JsCast::dyn_ref(&el): Option<&web_sys::Node> {
+                node.set_text_content(Some(&value));
+            }
+            self.el = Some(el);
+        }
+    }
+
+    /// Removes a class list item from the element
+    ///
+    /// ```
+    /// e.class_list_remove(String::from("clickable"));
+    /// // removes the class 'clickable' from e.el
+    /// ```
+    pub fn class_list_remove(&mut self, value: String) {
+        if let Some(el) = self.el.take() {
+            el.class_list().remove_1(&value);
+            self.el = Some(el);
+        }
+    }
+
+    /// Given another `Element` it will remove that child from the DOM from this element
+    pub fn remove_child(&mut self, mut child: Element) {
```

Comment:
This is another case where it could be `child: &Element`, but it depends on the goals of this I think (taking ownership is a Rust-ism which make more sense here)

Reply:
This was on purpose so it would consume `child` such that it isn't usable. I will add a comment that this is on purpose.